### PR TITLE
fix(@clayui/core): updates the sort icon for table

### PR DESCRIPTION
**Pull request by matuzalemsteles**
Closes [LPS-202866](https://liferay.atlassian.net/browse/LPS-202866)

The icon update is still in draft, I need to make some adjustments, @pat270, would you have any ideas on how we can try to achieve these use cases of the sorting icon state? I don't know if we have something like this in CSS.

Even though I'm using the button, I'm making it invisible to SR because we're using the head for sorting for accessibility reasons, so when you press the whole head it changes the sorting if it's enabled for the column, so we can use `aria -sort`.

### CODE REVIEW

**Comment by pat270** [comment]
@matuzalemsteles can we use a `span` tag for the icon and put `tabindex="0"` on `th` with a click event listener? I'm not sure if it breaks accessibility rules.

In my old examples, sorting in tables is using up or down arrow depending on the state

sort-icon

I can't find the new designs in the issue.

**Comment by ethib137** [comment]
The designs were attached to the parent ticket: https://www.figma.com/file/c5DKGFa5SSkf6a6se0rJrg/LPS-196505-Clay-Table-Styles-and-Variations?type=design&node-id=19-5188&mode=design&t=wzrbh8sc8xLP05Sm-0

**Comment by matuzalemsteles** [comment]
Actually now I'm a little confused the confluence document shows a different Figma icon https://liferay.atlassian.net/wiki/spaces/PEDS/pages/2541093048/Header+Rows+Interactions.

@emiliano-cicero can you tell which one we should follow here?

**Comment by ethib137** [comment]
@matuzalemsteles I think I must be missing something. It looks to me like [confluence](https://liferay.atlassian.net/wiki/spaces/PEDS/pages/2541093048/Header+Rows+Interactions) simply includes a screenshot of the [figma](https://www.figma.com/file/c5DKGFa5SSkf6a6se0rJrg/LPS-196505-Clay-Table-Styles-and-Variations?type=design&node-id=19-5188&mode=design&t=MyV1FNoIbHv5hSeT-0) documentation. What looks different to you?

**Comment by matuzalemsteles** [comment]
Oh @ethib137 now it looks ok, when I looked at the icon in the table it is using `order-list-down` and `order-list-up`.

**Comment by matuzalemsteles** [comment]
I went ahead and made some changes, I followed @pat270's idea of changing the button to span to move the focus to the `th` the idea was that but the focus went to the button which is bad for accessibility here because we need do focus on `th` so that SR announces the change using `aria-sort`. This is being done now it is possible to focus on the column in the head when treegrid is not enabled when the treegrid is enabled the focus is on the management of the first focusable element and it is possible to navigate to the head using the arrow keys, move the focus first to the cell and then navigate to the head.

I also added the CSS implementation to change the icon color depending on the sort. @pat270 let me know if that would be the best solution.

**Comment by pat270** [comment]
> I followed @pat270's idea of changing the button to span to move the focus to the th the idea was that but the focus went to the button which is bad for accessibility here because we need do focus on th so that SR announces the change using aria-sort.


@matuzalemsteles @ethib137 @marcoscv-work  I dug a little deeper into the sortable table accessibility patterns and came upon https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/.

Problems with placing tab focus on `th`:
1) The screenreader doesn't announce we can interact with the `th`. We need `role="button"`.
2) Adding `role="button"` on the `th` clobbers any announcement related to the `th`.
3) In the w3 sortable-table pattern, there is no requirement for announcing `aria-sort` when navigating with tab. 

The announcements are correct when navigating with `ctrl + option + arrows` / `ctrl + alt + arrows`. I think we should revert back to a button for the order arrows with an `aria-label` indicating it changes the order for the table column.

**Comment by ethib137** [comment]
Thanks @pat270 [this pattern](https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/sortable-table/) makes sense to me. @marcoscv-work does that look right to you? If so, @pat270 can you go ahead and make those updates?

**Comment by pat270** [comment]
@ethib137 Thanks! I'll implement if it's all good with @marcoscv-work. Last thing, do you think we could check if `Liferay.Language.get('what-we-want-to-announce-for-sorting')` exists if not default to something like `aria-label="What we want to to announce for sorting"`?

**Comment by ethib137** [comment]
> Last thing, do you think we could check if Liferay.Language.get('what-we-want-to-announce-for-sorting') exists if not default to something like aria-label="What we want to to announce for sorting"?
No we don't want to depend on Liferay. This is the pattern we normally use. We provide the label in English as a default value for a prop.

**Comment by pat270** [comment]
I added the `sortAriaLabel` attribute for custom announcements on the sort button.

**Comment by pedro-michel** [comment]
Any news here @pat270 ?

File: packages/clay-core/src/table/Cell.tsx [hunk 1]
```diff
@@ -92,6 +97,7 @@ export const Cell = React.forwardRef<HTMLTableCellElement, Props>(
 			expanded,
 			index,
 			keyValue,
+			sortAriaLabel = 'Sort Column',
```
Comment (matuzalemsteles) [inline packages/clay-core/src/table/Cell.tsx:100]: We're building a standard to use a single API called `messages`, I'm going to move this there so I can just add it in one place.

https://github.com/liferay/clay/blob/c5675c0443b4ec25273a4172589e3656cdaabfd7/packages/clay-core/src/table/Table.tsx#L54

**Pull request merged**